### DISCUSSION TITLE
Why do I have to add a type to setErrors?

### DISCUSSION THREAD

**Discussion by h0jeZvgoxFepBQ2C** [react-hook-form/react-hook-form#2697] [General]
As the title says: Why do I have to add a type to ```setErrors``` ?

I never use any local javascript validation schema, but only do server side validations. Every time I use react-hook-form it bugs me that I have to add a type.

It would be much better to just say:

```javascript
setError("firstname", "must be present")
```

Change my mind :D

**Comment by bluebill1049** [comment]
what if you have multiple errors associated with different types?

**Reply by chapterjason** [reply]
```
setError("firstname", "must be present")
setErrors("firstname", ["must be present", "must be at least 2 chars long"])
```
or as overload
```
setErrors("firstname", "must be present")
setErrors("firstname", ["must be present", "must be at least 2 chars long"])
```
For me the type is also not important, it's just to render the messages.

// Edit
I also have to add that it should be easy to add errors, without knowing some kind of object format.

**Reply by h0jeZvgoxFepBQ2C** [reply]
I also never use the types 👍 I only add them just because it's required, but I don't understand & use them.

**Comment by TroyWolf** [comment]
I agree that the `type` property for `setError()` is not very clear what it is about. I am simply not specifying a type and things seem to be working. So I'm not finding it's required. `"react-hook-form": "^6.7.2",`

For example, if my submission fails completely--like the API is not even running--I set an overall "form" error like so:
```
      setError("form", {
        message: "Something went wrong!"
      })
```
